openadmet/toolkit/filtering/physchem_filters.py

```diff
@@ -0,0 +1,237 @@
+    min_unit_sep : float
+        The minimum unit separation between pKa values (default is 1).
+    """
+    min_pka: float = Field(default=None, description="The minimum pKa value for the range check.")
+    max_pka: float = Field(default=None, description="The maximum pKa value for the range check.")
+    min_unit_sep: float = Field(default=None, description="The minimum unit separation between pKa values.")
+
+    def filter(self, df: pd.DataFrame, pka_column: str = "pka", mode="mark") -> pd.DataFrame:
+        """
+        Run the pKa filter on the DataFrame.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            The input DataFrame to be filtered.
+        mode : str
+            Either "mark" or "remove". If "mark", the filter will mark the rows that meet the criteria
+            either True or False. If "remove", the filter will remove the rows that meet the criteria.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The filtered DataFrame.
+        """
+        # check if the pka column exists
+        if pka_column not in df.columns:
+            raise ValueError(f"The DataFrame does not contain a {pka_column} column.")
+
+        if self.min_pka and self.max_pka:
+            # filter for at least one pka between min_pka and max_pka
+            df["pka_in_range"] = df["pka"].apply(lambda x: self.pkas_valid_range(x))
+
+        if self.min_unit_sep:
+            # filter for pka values that are at least min_unit_sep apart
+            df["pka_unit_sep"] = df["pka"].apply(lambda x : self.pka_separation(x, self.min_unit_sep))
+
+        return mark_or_remove(df, mode, ["pka_in_range", "pka_unit_sep"])
+
+    def pkas_valid_range(self, pkas: list) -> bool:
+        """
+        Check if the pKa values are within the specified range.
+
+        Parameters
+        ----------
+        pkas : list
+            A list of pKa values to be checked.
+
+        Returns
+        -------
+        bool
+            True if all pKa values are within the specified range.
+        """
+        valid_range = False
+        for pka in pkas:
+            if self.min_pka <= pka <= self.max_pka:
+                valid_range = True
+                break
+        return valid_range
+
+    def pka_separation(self, pkas: list, min_unit_sep: float) -> bool:
+        """
+        Check if the pKa values are at least min_unit_sep apart.
+
+        Parameters
+        ----------
+        pkas : list
+            A list of pKa values to be checked.
+        min_unit_sep : float
+            The minimum unit separation between pKa values.
+
+        Returns
+        -------
+        bool
+            True if all pKa values are at least min_unit_sep apart.
+        """
+        for i in range(len(pkas)):
+            for j in range(i + 1, len(pkas)):
+                if abs(pkas[i] - pkas[j]) < min_unit_sep:
+                    return False
+        return True
+
+class logPFilter(BaseFilter):
```

Comment:
Lets leave this one out as I think covered by something like: https://docs.datamol.io/0.7.4/api/datamol.descriptors.html and can be handled downstream.